query/src/main/java/org/infinispan/query/impl/IndexStartupActivator.java

```diff
@@ -0,0 +1,56 @@
+package org.infinispan.query.impl;
+
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.IndexStartupMode;
+import org.infinispan.configuration.cache.IndexStorage;
+import org.infinispan.query.Indexer;
+import org.infinispan.search.mapper.mapping.SearchMapping;
+
+public final class IndexStartupActivator {
+
+   private final SearchMapping mapping;
+   private final Indexer indexer;
+   private final Configuration configuration;
+
+   public static IndexStartupActivator create(SearchMapping mapping, Indexer indexer, Configuration configuration) {
+      return new IndexStartupActivator(mapping, indexer, configuration);
+   }
+
+   private IndexStartupActivator(SearchMapping mapping, Indexer indexer, Configuration configuration) {
+      this.mapping = mapping;
+      this.indexer = indexer;
+      this.configuration = configuration;
+   }
+
+   public void run() {
+      IndexStartupMode startupMode = computeFinalMode();
+
+      if (IndexStartupMode.PURGE.equals(startupMode)) {
```

Comment:
IIUC reading [the answers here](https://stackoverflow.com/questions/1750435/comparing-java-enum-members-or-equals) the benefits would be to:
1. null safety of `==`, but in this case we used (CompareConstantsFromTheLeft)[http://wiki.c2.com/?CompareConstantsFromTheLeft].
2. check at compile time, but in this case we have a variable to compare with.

I'm not really about the fact that this could be an an improvement.

Reply:
from the answers, it seems a personal preference :)